mindsdb/integrations/handlers/weaviate_handler/weaviate_handler.py

```diff
@@ -0,0 +1,617 @@
+                        value_type: condition.value,
+                    }
+                )
+        # condition on metadata columns
+        if meta_conditions is not None:
+            for condition in meta_conditions:
+                meta_key = condition.column.split(".")[-1]
+                value_type = self._get_weaviate_value_type(condition.value)
+                weaviate_conditions.append(
+                    {
+                        "path": [
+                            "associatedMetadata",
+                            metadata_table_name,
+                            meta_key,
+                        ],
+                        "operator": self._get_weaviate_operator(condition.op),
+                        value_type: condition.value,
+                    }
+                )
+
+        # we combine all conditions into a single dict
+        all_conditions = (
+            {"operator": "And", "operands": weaviate_conditions}
+            if len(weaviate_conditions) > 1
+            else weaviate_conditions[0]
+        )
+        return all_conditions
+
+    def select(
+        self,
+        table_name: str,
+        columns: List[str] = None,
+        conditions: List[FilterCondition] = None,
+        offset: int = None,
+        limit: int = None,
+    ) -> HandlerResponse:
+        table_name = table_name.capitalize()
+        # columns which we will always provide in the result
+        fixed_columns = ["id", "embeddings", "distance", "metadata"]
+        filters = None
+        if conditions is not None:
+            non_metadata_conditions = [
+                condition
+                for condition in conditions
+                if not condition.column.startswith(TableField.METADATA.value)
+                and condition.column != TableField.SEARCH_VECTOR.value
+                and condition.column != TableField.EMBEDDINGS.value
+            ]
+            metadata_conditions = [
+                condition
+                for condition in conditions
+                if condition.column.startswith(TableField.METADATA.value)
+            ]
+            filters = self._translate_condition(
+                table_name,
+                non_metadata_conditions if len(non_metadata_conditions) > 0 else None,
+                metadata_conditions if len(metadata_conditions) > 0 else None,
+            )
+
+        # check if embedding vector filter is present
+        vector_filter = (
+            []
+            if conditions is None
+            else [
+                condition
+                for condition in conditions
+                if condition.column == TableField.SEARCH_VECTOR.value
+                or condition.column == TableField.EMBEDDINGS.value
+            ]
+        )
+
+        if len(vector_filter) > 0:
+            vector_filter = vector_filter[0]
+        else:
+            vector_filter = None
+
+        for col in fixed_columns:
```

Comment:
for col in ("id", "embeddings", "distance", "metadata")

Reply:
resolved in b67cb1454fe8803bc304df7430cab6bcd5d670e7